src/Orleans.Core/Serialization/OrleansJsonSerializer.cs

```diff
@@ -9,22 +9,31 @@
 
 namespace Orleans.Serialization
 {
+    using Microsoft.Extensions.Options;
     using Orleans.Providers;
 
     public class OrleansJsonSerializer : IExternalSerializer
     {
         public const string UseFullAssemblyNamesProperty = "UseFullAssemblyNames";
         public const string IndentJsonProperty = "IndentJSON";
         public const string TypeNameHandlingProperty = "TypeNameHandling";
+        private readonly OrleansJsonSerializerOptions jsonSerializerOptions;
         private readonly Lazy<JsonSerializerSettings> settings;
 
-        public OrleansJsonSerializer(IServiceProvider services)
+        public OrleansJsonSerializer(
+            IServiceProvider services,
+            IOptions<OrleansJsonSerializerOptions> jsonSerializerOptions)
         {
+            this.jsonSerializerOptions = jsonSerializerOptions.Value;
+
             this.settings = new Lazy<JsonSerializerSettings>(() =>
             {
                 var typeResolver = services.GetRequiredService<ITypeResolver>();
                 var grainFactory = services.GetRequiredService<IGrainFactory>();
-                return GetDefaultSerializerSettings(typeResolver, grainFactory);
+                JsonSerializerSettings settings = GetDefaultSerializerSettings(typeResolver, grainFactory);
+
+                this.jsonSerializerOptions.ConfigureSerializer(settings);
```

Comment:
I undesrtand why it was done like this, but it's a bit messy IMO. It would be better if the `JsonSerializerSettings` was configured directly in `OrleansJsonSerializerOptions` instead of having this list of actions.

Reply:
I wasn't entirely sure about this either, but my concern with that was that it would overwrite the default settings provided by `GetDefaultSerializerSettings`. Did you have another way in mind that would avoid that?

 Since `GetDefaultSerializerSettings` requires a type resolver and grain factory instance, it's not possible to call during silo configuration time as far as I can tell.